cpp/cmake/thirdparty/get_nvtx.cmake

```diff
@@ -0,0 +1,30 @@
+# =============================================================================
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+# This function finds NVTX and sets any additional necessary environment variables.
+function(find_and_configure_nvtx)
+  rapids_cpm_find(
+    nvtx 3.1.0
+    GIT_REPOSITORY https://github.com/NVIDIA/NVTX.git
+    GIT_TAG v3.1.0
+    GIT_SHALLOW TRUE
+    DOWNLOAD_ONLY TRUE
+  )
+  set(NVTX_INCLUDE_DIR
```

Comment:
I get the feeling that this is not the right way to do this. I think I need to add the nvtx targets, rather than specifying the include directory directly.

The targets are defined by `c/CMakeLists.txt` in the nvtx repo. I googled around a bit and tried to find how to add this. Do I need something like `SOURCE_SUBDIR c` for it to find the targets? I'm not sure how to use this.

cc: @robertmaynard @vyasr

Reply:
> Do I need something like SOURCE_SUBDIR c for it to find the targets? 

Yes 🙂 

https://github.com/NVIDIA/NVTX/tree/dev/c#use-cmake-package-manager-cpm

Reply:
Hmm. I tried this but wasn’t having success. I think that there is something I don’t understand about how rapids-cmake calls CPM.

Reply:
You need to use `CPM _ARGS SOURCE_SUBDIR c` to tell rapids-cmake to pass the option down to cpm.

Reply:
@robertmaynard I think I did that in 226101b. In my understanding, that should make the `nvtx3-cpp` target available, and I should be able to use `target_link_libraries(cudf PRIVATE nvtx3-cpp)` to get it to recognize the include directories. However, this isn't working as I expected. Can you help me debug it?